### PR TITLE
Fix/1039 Disable withdraw button if ethereum address is invalid

### LINKED ISSUE #1039: Disable Withdrawal button when incorrect Eth Address is given in the TO field
1) Attempt to initiate a withdrawal 
2) Click on Enter address manually option on To field on withdrawal page
3) Input a silly Eth address manually in text field
4) Click on the Withdrawal Vega Token Button
5) Observe the page

Actual -
It gives me error but Withdrawal Vega Token Button is enabled. However clicking on Withdrawal Vega Token doesn't do anything. IMO the button should be disabled when input isn't correct

Screenshot 2021-12-03 at 14 26 39

### PR DESCRIPTION
**Pull request by pgriscti**
closes #1039

### CODE REVIEW

File: src/routes/withdraw/withdraw-form.tsx [hunk 1]
```diff
@@ -69,6 +69,11 @@ export const WithdrawForm = ({
     }
   }, [status, history]);
 
+  React.useMemo(async () => {
```
Comment (mattrussell36) [inline src/routes/withdraw/withdraw-form.tsx:72]: You are using useMemo like a useEffect

Comment (mattrussell36) [reply src/routes/withdraw/withdraw-form.tsx:72]: Also, I'm not sure we need to call ethers.utils.isAddress in two places. Can we have a single useMemo for `addressValid`? I'm not sure we need an additional useState for storing its validity.

Then we can return early in on submit if `!addressValid` and additionally prevent submit at all by disabling the button?

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

File: src/routes/withdraw/withdraw-form.tsx (hunk 1 above)
Comment (pgriscti) [reply src/routes/withdraw/withdraw-form.tsx]: changed

File: src/routes/withdraw/withdraw-form.tsx [hunk 2]
```diff
@@ -69,13 +69,19 @@ export const WithdrawForm = ({
     }
   }, [status, history]);
 
+  React.useEffect(() => {
+    const run = async () => {
+      await setAddressValid(ethers.utils.isAddress(destinationAddress));
```
Comment (mattrussell36) [inline src/routes/withdraw/withdraw-form.tsx:74]: You're awaiting on nothing here because the state setter doesn't return anything.

Comment (mattrussell36) [reply src/routes/withdraw/withdraw-form.tsx:74]: Also ethers.utils.isAddress doesn't return a promise at all so theres no need to use an async function here

**Review by mattrussell36 — CHANGES_REQUESTED** [review]

**Pull request merged**